src/WebJobs.Script/Workers/Rpc/Configuration/DynamicWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    // This class resolves worker configurations dynamically based on the current environment and configuration settings.
+    // It searches for worker configs in specified probing paths and the fallback path, and returns a list of worker configuration paths.
+    internal sealed class DynamicWorkerConfigurationResolver : IWorkerConfigurationResolver
+    {
+        private readonly IConfiguration _config;
+        private readonly ILogger _logger;
+        private readonly IWorkerProfileManager _profileManager;
+        private readonly IEnvironment _environment;
+        private readonly IFileSystem _fileSystem;
+        private readonly HashSet<string> _workersAvailableForResolutionViaHostingConfig;
+        private readonly List<string> _workerProbingPaths;
+        private readonly JsonSerializerOptions _jsonSerializerOptions = new() { PropertyNameCaseInsensitive = true };
+
+        public DynamicWorkerConfigurationResolver(IConfiguration config,
+                                        ILogger logger,
+                                        IEnvironment environment,
```

Comment:
Here's another example -- using IEnvironment here doesn't communicate exactly what we need. But having some kind of IOptions<> (or multiple if some of the values exist on options elsewhere already) can make it clearer.